### PR TITLE
Add main sequence diagram for arbitrating auctions

### PR DESCRIPTION
**Pull request by sunce86**
# Description

Implements first part of the https://github.com/cowprotocol/docs-v2/issues/238:

> Could we add a diagram that shows the user, the orderbook, the autopilot and solvers (without differentiating between driver and engines)?

### CODE REVIEW

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: We should probably use a different word than `solution` because the next call fetches the "Solution with calldata" which begs the question why we split it into 2 calls in the first place.
```suggestion
    solver-->>autopilot: Score
```

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]:
```suggestion
    orderbook-->>autopilot: Response solvable orders
```

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: All the other endpoints actually exist but not this one.
```suggestion
    User->>orderbook: POST /order
```

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: Should we also show the DB here since the autopilot is actually just talking to that instead of the orderbook or would that go into too much detail?
Not sure what level of detail we want to have here.

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: Thief! 😄 
To make this easier to run we can also consider writing the config files inside `run.sh`. Then the user only has to copy and paste a single file.
Not sure what is nicer for a tutorial, though.

Comment (sunce86) [reply docs/cow-protocol/tutorials/arbitrate/README.mdx]: gollum

Comment (sunce86) [reply docs/cow-protocol/tutorials/arbitrate/README.mdx]: cat

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: Would be nice to add the name of the tokens as a comment.

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: We can drop the token owners cache thing.
I only had it in my example because the yearn team needed some help setting this up and they relied on this feature.

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: I believe we actually no longer have this argument. 😅

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (fleupold) [reply docs/cow-protocol/tutorials/arbitrate/README.mdx]: Since the text is talking about three main components, I actually like that the diagram focusses on that. Will move the diagram up to the architecture section.

**Review by fleupold — COMMENTED** [review]
I know I was asking for the "run" instructions, but I'm worried those will outdate faster than we can type whitespaces. @mfw78 do you have a suggestion on how we can refer to a place where developers can see an up to date instruction on how to run the services? Maybe the readme or the devcontainer file you started to work on?

**Comment by fleupold** [comment]
We have it in version control, but I removed the detailed diagram as it doesn't maintain the driver/solver subdifferentiation and we have that information in the sections below. Happy to put it back if you feel it was better before.

For running locally it would be nice to link to instructions on how to run both the "shadow mode" as well as an oderbook + system (eventually also with the UI running)

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (sunce86) [reply docs/cow-protocol/tutorials/arbitrate/README.mdx]: Added.

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (sunce86) [reply docs/cow-protocol/tutorials/arbitrate/README.mdx]: Removed.

**Comment by sunce86** [comment]
> I know I was asking for the "run" instructions, but I'm worried those will outdate faster than we can type whitespaces.

I believe this simplified example is probably most important among all tutorials and should be kept up to date (and probably be part of the services readme as you suggested). One thing we could do is to update the `services` PR template and add the acceptance tests part where the "check if docs need to be updated" would need to be verified before merging (although error prone but better than nothing).

**Comment by mfw78** [comment]
> > I know I was asking for the "run" instructions, but I'm worried those will outdate faster than we can type whitespaces.
> 
> I believe this simplified example is probably most important among all tutorials and should be kept up to date (and probably be part of the services readme as you suggested). One thing we could do is to update the `services` PR template and add the acceptance tests part where the "check if docs need to be updated" would need to be verified before merging (although error prone but better than nothing).

I'm currently pulling this all into a `docker-compose` playground for integration within the `services` repo.

**Review by fleupold — APPROVED** [review]
LGTM, leaving it up to @mfw78 to weigh on if we want to keep the run instructions (which are quite detailed and maybe a link to instructions that will be more reasonably kept up to date may make more sense)

File: docs/cow-protocol/tutorials/arbitrate/README.mdx
Comment (MartinquaXD) [inline docs/cow-protocol/tutorials/arbitrate/README.mdx]: BTW the `_SECONDS` arguments will be outdated when https://github.com/cowprotocol/services/pull/2194 is merged.

**Review by MartinquaXD — APPROVED** [review]

**Review by mfw78 — APPROVED** [review]
I've removed the run script and driver config files and made a note on https://github.com/cowprotocol/services/pull/2228 that pointing to the playground readme needs to be documented here when it's pushed.

**Pull request merged**
